src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -351,6 +355,45 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+
+        /// <summary>
+        /// Handles a request to rename the activequery and return the result
+        /// </summary>
+        internal async Task HandleRenameRequest(QueryRenameParams renameParams,
+            RequestContext<QueryRenameResult> requestContext)
+        {
+            try
+            {
+                string OriginalOwnerUri = renameParams.OriginalOwnerUri;
+                string NewOwnerUri = renameParams.NewOwnerUri;
+                // Attempt to load the query
+                Query query;
+                if (!ActiveQueries.TryGetValue(OriginalOwnerUri, out query))
+                {
+                    await requestContext.SendError(SR.QueryServiceRequestsNoQuery);
+                    return;
+                }
+                else
+                {
+                    ActiveQueries.TryRemove(OriginalOwnerUri, out query);
+                    // Delete the map from ConnectionService uri connection map, reconnect with new uri later.
+                    ConnectionService.Disconnect(new DisconnectParams()
```

Comment:
Why are you disconnecting the connection? A connection is independent of the editor that opened it - if we rename a file we should update the URI but the connection should stay open as it was.